src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -3558,150 +3558,154 @@ instruct vsqrt_fp_masked(vReg dst_src, vRegMask_V0 v0) %{
   ins_pipe(pipe_slow);
 %}
 
-instruct vstring_equalsL(iRegP_R11 str1, iRegP_R13 str2, iRegI_R14 cnt,
-                         iRegI_R10 result, vReg_V2 v2,
-                         vReg_V3 v3, vReg_V4 v4, vReg_V5 v5, rFlagsReg cr)
+instruct vstring_equalsL(iRegP_R11 str1, iRegP_R13 str2, iRegI_R14 cnt, iRegI_R10 result,
+                         vReg_V2_m2 v2m2, vReg_V4_m2 v4m2, rFlagsReg cr)
 %{
   predicate(UseRVV && ((StrEqualsNode*)n)->encoding() == StrIntrinsicNode::LL);
   match(Set result (StrEquals (Binary str1 str2) cnt));
-  effect(USE_KILL str1, USE_KILL str2, USE_KILL cnt, TEMP v2, TEMP v3, TEMP v4, TEMP v5, KILL cr);
+  effect(USE_KILL str1, USE_KILL str2, USE_KILL cnt, TEMP v2m2, TEMP v4m2, KILL cr);
 
   format %{ "String Equals $str1, $str2, $cnt -> $result\t#@string_equalsL" %}
   ins_encode %{
     // Count is in 8-bit bytes; non-Compact chars are 16 bits.
     __ string_equals_v($str1$$Register, $str2$$Register,
-                       $result$$Register, $cnt$$Register);
+                       $result$$Register, $cnt$$Register,
+                      $v2m2$$VectorRegisterGroup, $v4m2$$VectorRegisterGroup);
```

Comment:
I also like that we pass a single "register" instead of of `v2` and `v3` separately. I also like we pass them explicitly here so that there is not the implicit knowledge that `string_equals_v` uses `v2` and `v3` with a `lmul=m2` internally.